core/pkg/service/sync/handler.go

```diff
@@ -0,0 +1,59 @@
+package sync
+
+import (
+	"context"
+
+	syncv1 "buf.build/gen/go/open-feature/flagd/protocolbuffers/go/sync/v1"
+	"github.com/bufbuild/connect-go"
+	"github.com/open-feature/flagd/core/pkg/logger"
+	"github.com/open-feature/flagd/core/pkg/sync"
+	syncStore "github.com/open-feature/flagd/core/pkg/sync-store"
+)
+
+type handler struct {
+	syncStore *syncStore.SyncStore
+	logger    *logger.Logger
+}
+
+func (l *handler) FetchAllFlags(ctx context.Context, req *connect.Request[syncv1.FetchAllFlagsRequest]) (
+	*connect.Response[syncv1.FetchAllFlagsResponse],
+	error,
+) {
+	data, err := l.syncStore.FetchAllFlags(ctx, nil, req.Msg.GetProviderId())
```

Comment:
And let's utilize `filter` field instead of `providerId` [1]. This makes the implementation generic :) 

[1] - https://github.com/open-feature/schemas/pull/89